packages/frontend/src/pages/Tiles/components/CreateTileButton.tsx

```diff
@@ -1,62 +1,199 @@
 import { ITableMetadata } from '@plumber/types'
 
-import { FormEvent, useCallback, useContext, useState } from 'react'
+import { useCallback, useContext, useState } from 'react'
+import { BiSpreadsheet, BiTable } from 'react-icons/bi'
 import { HiOutlineHandRaised } from 'react-icons/hi2'
 import { useNavigate } from 'react-router-dom'
 import { useMutation } from '@apollo/client'
 import {
+  Flex,
   Modal,
   ModalBody,
   ModalContent,
   ModalFooter,
   ModalHeader,
   ModalOverlay,
+  Text,
   useDisclosure,
 } from '@chakra-ui/react'
 import {
   Button,
   FormLabel,
   Input,
   ModalCloseButton,
+  Tile,
 } from '@opengovsg/design-system-react'
 import { TILES_FEATURE_FLAG } from 'config/flags'
 import * as URLS from 'config/urls'
 import { LaunchDarklyContext } from 'contexts/LaunchDarkly'
 import { CREATE_TABLE } from 'graphql/mutations/create-table'
+import { ImportCsvModalContent } from 'pages/Tile/components/TableBanner/ImportCsvButton'
+import { TableContextProvider } from 'pages/Tile/contexts/TableContext'
 
-const CreateTileButton = (): JSX.Element => {
-  /**
-   * Check if feature flag is enabled, otherwise dont allow creation
-   * eventually, this should be removed
-   */
-  const { flags } = useContext(LaunchDarklyContext)
-  const isTilesEnabled = flags?.[TILES_FEATURE_FLAG]
+type TILE_CREATE_MODE = 'import' | 'new'
```

Comment:
nit: use an enum instead?

Reply:
https://www.google.com/search?q=enum+or+union+type&ie=UTF-8&oe=UTF-8&hl=en-sg&client=safari

unions are easier to work with